test/c/test_target.c

```diff
@@ -552,6 +553,90 @@ static int test_target_update_instruction(void) {
     return result;
 }
 
+/**
+ * Test if our target is compatible with certain instructions.
+ */
+int test_target_instruction_supported(void) {
+    QkTarget *sample_target = create_sample_target(true);
+    int result = Ok;
+
+    char *gate_names[7] = {"x", "y", "id", "rz", "sx", "reset"};
+    QkParam *rz_params[1] = {
+        qk_param_from_double(3.14),
+    };
+    for (uint32_t qubit = 0; qubit < 5; qubit++) {
+        uint32_t qargs[1] = {qubit};
+        bool should_be_true = qubit < 4;
+
+        for (int gate = 0; gate < 6; gate++) {
+            // If i == 4 condition should be false unless we try with the y gate
+            // since y is added as a global gate.
+            if (!(qk_target_instruction_supported(sample_target, gate_names[gate], qargs,
+                                                  gate != 3 ? NULL : rz_params) ==
+                  should_be_true) &&
+                gate != 1) {
```

Comment:
This doesn't properly validate that `y` is supported as a global instruction since it's short circuiting on `gate == 1`. 
We can refactor the condition to something like this:

```suggestion
            if (qk_target_instruction_supported(sample_target, gate_names[gate], qargs,
                                                  gate != 3 ? NULL : rz_params) !=
                  (should_be_true || gate == 1)) {
```